### PR TITLE
Add short-term storage doc for optim model

### LINKED ISSUE #2239: Optimization documentation missing short-term storage model
https://antares-simulator.readthedocs.io/en/stable/user-guide/solver/05-model/

See (internal link) https://gopro-collaboratif.rte-france.com/pages/viewpage.action?pageId=228787847

See also https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2223 for recent changes

### PR DESCRIPTION
**Pull request by flomnes**
Preview https://antares-simulator--2283.org.readthedocs.build/en/2283/user-guide/solver/05-model/#short-term-storage

- Notations (variables, parameters)
- Added level equation
- Contribution to the existing balance equation

### CODE REVIEW

File: docs/user-guide/solver/05-model.md [hunk 1]
```diff
@@ -178,6 +178,21 @@ Note: Almost all variables of the system are defined twice (one value per state)
 | $R\_\lambda \in \mathbb{R}^T_+$                   | stored energy level in reservoir $\lambda$                                                                                                                                         |
 | $\mathfrak{R}\_{\lambda_q} \in \mathbb{R}_+$      | filling level of reservoir layer $q$ at time $T$ (end of the week)                                                                                                                 |
 
+### Short-term storages
+| Notation                                          | Explanation                                                                                                                                                                        |
```
Comment (tbittar) [inline docs/user-guide/solver/05-model.md:182]: For all time varying data, either add subscript `t` in the notation or use the vector form as everywhere else in the notation tables

**Review by tbittar — CHANGES_REQUESTED** [review]

File: docs/user-guide/solver/05-model.md (hunk 1 above)
Comment (flomnes) [reply docs/user-guide/solver/05-model.md:182]: Fixed in 98f69b6d07079974714e7d8282e9abcb9081aae1

File: docs/user-guide/solver/05-model.md [hunk 2]
```diff
@@ -178,6 +178,21 @@ Note: Almost all variables of the system are defined twice (one value per state)
...
+| $L_s \in \mathbb{R}_+^T$      | Level for storage $s$ |
+| $\underline{L}_s \in \mathbb{R}^T$, $\overline{L}_s \in \mathbb{R}^T$    | Minimum (resp. maximum) level for storage $s$ also known as "rule-curves" |
+| $L_s^0 \in \mathbb{R}_+$     | Initial level for storage $s$ (optional) |
+| $P^w_s \in \mathbb{R}_+^T$    | Withdrawal for storage $s$. Note that this is from the storage's perspective : the amount of power withdrawn from the storage |
+| $\underline{P}^i_s \in \mathbb{R}^T$, $\overline{P}^i_s \in \mathbb{R}^T$    | Minimum (resp. maximum) injection at time $t$ for storage $s$ |
+| $\eta^i_s \in [0, 1]$ | Injection efficiency for storage $s$ |
+| $P^i_s \in \mathbb{R}_+^T$    | Injection for storage $s$. Note that this is from the storage's perspective : the amount of power injected into the storage |
+| $\underline{P}^w_s \in \mathbb{R}^T$, $\overline{P}^w_s \in \mathbb{R}^T$    | Minimum (resp. maximum) withdrawal at time $t$ for storage $s$ |
+| $\eta^w_s \in [0, 1]$ | Withdrawal efficiency for storage $s$ |
+| $I_s \in \mathbb{R}^T$   | Inflows for storage $s$ at time $t$. Energy that is injected into the storage over time |
```
Comment (tbittar) [inline docs/user-guide/solver/05-model.md]: Remove `at time t`

File: docs/user-guide/solver/05-model.md [hunk 3]
```diff
@@ -178,6 +178,21 @@ Note: Almost all variables of the system are defined twice (one value per state)
...
+| $s\in \mathcal{S}$      | A single short-term storge reservoir. There is one set of short-term storage per area. |
+| $L_s \in \mathbb{R}_+^T$      | Level for storage $s$ |
+| $\underline{L}_s \in \mathbb{R}^T$, $\overline{L}_s \in \mathbb{R}^T$    | Minimum (resp. maximum) level for storage $s$ also known as "rule-curves" |
+| $L_s^0 \in \mathbb{R}_+$     | Initial level for storage $s$ (optional) |
+| $P^w_s \in \mathbb{R}_+^T$    | Withdrawal for storage $s$. Note that this is from the storage's perspective : the amount of power withdrawn from the storage |
+| $\underline{P}^i_s \in \mathbb{R}^T$, $\overline{P}^i_s \in \mathbb{R}^T$    | Minimum (resp. maximum) injection at time $t$ for storage $s$ |
+| $\eta^i_s \in [0, 1]$ | Injection efficiency for storage $s$ |
+| $P^i_s \in \mathbb{R}_+^T$    | Injection for storage $s$. Note that this is from the storage's perspective : the amount of power injected into the storage |
+| $\underline{P}^w_s \in \mathbb{R}^T$, $\overline{P}^w_s \in \mathbb{R}^T$    | Minimum (resp. maximum) withdrawal at time $t$ for storage $s$ |
```
Comment (tbittar) [inline docs/user-guide/solver/05-model.md]: Remove `at time t`

File: docs/user-guide/solver/05-model.md [hunk 4]
```diff
@@ -178,6 +178,21 @@ Note: Almost all variables of the system are defined twice (one value per state)
...
+| Notation                                          | Explanation                                                                                                                                                                        |
+|---------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| $s\in \mathcal{S}$      | A single short-term storge reservoir. There is one set of short-term storage per area. |
+| $L_s \in \mathbb{R}_+^T$      | Level for storage $s$ |
+| $\underline{L}_s \in \mathbb{R}^T$, $\overline{L}_s \in \mathbb{R}^T$    | Minimum (resp. maximum) level for storage $s$ also known as "rule-curves" |
+| $L_s^0 \in \mathbb{R}_+$     | Initial level for storage $s$ (optional) |
+| $P^w_s \in \mathbb{R}_+^T$    | Withdrawal for storage $s$. Note that this is from the storage's perspective : the amount of power withdrawn from the storage |
+| $\underline{P}^i_s \in \mathbb{R}^T$, $\overline{P}^i_s \in \mathbb{R}^T$    | Minimum (resp. maximum) injection at time $t$ for storage $s$ |
```
Comment (tbittar) [inline docs/user-guide/solver/05-model.md]: remove `at time t`

**Review by tbittar — CHANGES_REQUESTED** [review]
One last effort :p

**Review by tbittar — APPROVED** [review]

**Pull request merged**